File: 技能库&准则/anydoc-main/examples/convert.mjs
// Convert a document to Markdown:
// `node examples/convert.mjs <file> [-f csv] [-o out.md] [--assets dir]`
//
// Run `npm install && npm run build` in `node/` first, or import
// '@firecrawl/anydoc' instead of the local build.
import { mkdir, readFile, writeFile } from 'node:fs/promises';
import { basename, extname, join } from 'node:path';
import { argv, exit, stdout } from 'node:process';

import {
  formatFromBytes,
  formatFromPath,
  toDocument,
  toMarkdownBytes,
} from '../node/index.js';

const USAGE = 'usage: convert <file> [-f csv] [-o out.md] [--assets dir]';

const millis = (ms) => (ms < 10 ? `${ms.toFixed(2)}ms` : `${ms.toFixed(0)}ms`);

const args = argv.slice(2);
let input;
let output;
let format;
let assets;
for (let i = 0; i < args.length; i++) {
  switch (args[i]) {
    case '-o':
    case '--output':
      output = args[++i];
      break;
    case '-f':
    case '--format':
      format = args[++i];
      break;
    case '--assets':
      assets = args[++i];
      break;
    default:
      input = args[i];
  }
}
if (!input) {
  console.error(USAGE);
  exit(1);
}

try {
  const bytes = await readFile(input);
  // Without -f the format comes from the file content, with the extension as
  // the fallback.
  format ??= formatFromBytes(bytes) ?? formatFromPath(input);
  if (!format) {
    throw new Error(`unrecognized file content and extension: ${input}`);
  }

  const start = performance.now();
  const markdown = await toMarkdownBytes(bytes, format);
  const elapsed = performance.now() - start;
  console.error(`converted ${input} in ${millis(elapsed)}`);

  if (output) {
    await writeFile(output, markdown);
  } else {
    stdout.write(markdown);
  }

  // Images and embedded objects live on the document model, not in the
  // Markdown, so they need a second pass to write out.
  if (assets) {
    const document = await toDocument(bytes, format);
    await mkdir(assets, { recursive: true });
    const stem = basename(input, extname(input));
    for (const asset of document.assets) {
      const [kind, subtype] = asset.mediaType.split('/');
      const extension =
        kind === 'image' ? subtype.replace(/[^a-z0-9]/gi, '') : 'bin';
      await writeFile(
        join(assets, `${stem}-${asset.id}.${extension}`),
        asset.data,
      );
    }
    console.error(`wrote ${document.assets.length} assets to ${assets}`);
  }
} catch (error) {
  console.error(`error: ${error.message}`);
  exit(1);
}

File: 技能库&准则/anydoc-main/examples/convert.py
# Convert a document to Markdown:
# `python examples/convert.py <file> [-f csv] [-o out.md] [--assets dir]`
#
# Run `pip install maturin && maturin develop` in `python/` first, or
# `pip install firecrawl-anydoc` instead of the local build.
import argparse
import re
import sys
import time
from pathlib import Path

import anydoc


def millis(ms: float) -> str:
    return f"{ms:.2f}ms" if ms < 10 else f"{ms:.0f}ms"


parser = argparse.ArgumentParser(prog="convert", add_help=False)
parser.add_argument("file")
parser.add_argument("-f", "--format")
parser.add_argument("-o", "--output")
parser.add_argument("--assets")
args = parser.parse_args()

try:
    data = Path(args.file).read_bytes()
    # Without -f the format comes from the file content, with the extension
    # as the fallback.
    format = args.format or anydoc.format_from_bytes(data) or anydoc.format_from_path(args.file)
    if not format:
        raise ValueError(f"unrecognized file content and extension: {args.file}")

    start = time.perf_counter()
    markdown = anydoc.to_markdown_bytes(data, format)
    elapsed = (time.perf_counter() - start) * 1000
    print(f"converted {args.file} in {millis(elapsed)}", file=sys.stderr)

    if args.output:
        Path(args.output).write_text(markdown, encoding="utf-8")
    else:
        sys.stdout.write(markdown)

    # Images and embedded objects live on the document model, not in the
    # Markdown, so they need a second pass to write out.
    if args.assets:
        document = anydoc.to_document(data, format)
        assets = Path(args.assets)
        assets.mkdir(parents=True, exist_ok=True)
        stem = Path(args.file).stem
        for asset in document.assets:
            kind, _, subtype = asset.media_type.partition("/")
            extension = re.sub(r"[^a-z0-9]", "", subtype, flags=re.I) if kind == "image" else "bin"
            (assets / f"{stem}-{asset.id}.{extension}").write_bytes(asset.data)
        print(f"wrote {len(document.assets)} assets to {args.assets}", file=sys.stderr)
except Exception as error:
    print(f"error: {error}", file=sys.stderr)
    sys.exit(1)
